src/io.c

```diff
@@ -137,6 +137,48 @@ static int pids[MAXCHLDS];         /* and this number! */
 static int fistats = 0;            /* First used entry */
 static int lastats = 0;            /* First unused entry */
 static int statsfull = 0;          /* Flag, whether stats FIFO full */
+
+static int findSignaledPid(int pidc)
```

Comment:
Please add a comment that warns that this function must only be called if our child signal handler is disabled.